cassovary-core/src/main/scala/com/twitter/cassovary/graph/Node.scala

```diff
@@ -273,9 +289,15 @@ trait Node {
 
 object Node {
   private lazy val randGen: Random = new Random
-  def apply(nodeId: Int, out: Seq[Int], in: Seq[Int]) = new Node {
-    val id = nodeId
-    def inboundNodes() = in
-    def outboundNodes() = out
+
+  def apply(nodeId: Int, out: Seq[Int], in: Seq[Int]): Node = {
+    new SeqBasedNode(nodeId, in, out)
+  }
+
+  def withSortedNeighbors(nodeId: Int, in: Array[Int], out: Array[Int]) = {
+    new SeqBasedNode(nodeId, in, out) with SortedNeighborsNodeOps
   }
 }
+
+class SeqBasedNode private[graph] (val id: Int, val inboundNodes: Seq[Int], val outboundNodes: Seq[Int])
```

Comment:
No, because of other methods. It is a bit more general. Consider:
```
class SetBasedNode(val id, val in: Set[Int], val out: Set[Int]) extends Node {
  def inboundNodes() = in.toSeq // slower
  def isInboundNode(node: Int) = ... // faster
  ...
}
```
You cannot do this using abstract class Node based on seqs.